eclipse-collections-code-generator/src/main/resources/api/block/function/primitiveFunction.stg

```diff
@@ -22,7 +22,7 @@ import java.io.Serializable;
  *
  * @since 3.0.
  */
-<if(primitive.specializedStream)>@FunctionalInterface<endif>
+@FunctionalInterface
```

Comment:
This one may be a difference between Sonar static analysis and IntelliJ static analysis. In IntelliJ, go to the folder containing generated sources. Right click the folder and select unmark as generated source root. It will temporarily show up like regular source, so inspections will run. I suspect that IntelliJ will warn you that the @FunctionalInterface annotation is redundant and ask you to remove it. It is redundant, because it's on the parent interface. In cases where static analysis conflicts, we just have to pick a rule and be consistent. I tend to prefer IntelliJ rules because they make more use of the type system to eliminate redundancy. But there's an argument to be made that the increased redundancy helps readability without jumping between files.